modules/solr/src/main/java/org/testcontainers/containers/SolrClientUtils.java

```diff
@@ -0,0 +1,124 @@
+
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * Utils class which can create collections and configurations.
+ *
+ * @author Simon Schneider
+ */
+public class SolrClientUtils {
+
+    private static OkHttpClient httpClient = new OkHttpClient();
+
+    /**
+     * Creates a new configuration and uploads the solrconfig.xml and schema.xml
+     *
+     * @param port              the Port on which solr is running
+     * @param configurationName the name of the configuration which should be created
+     * @param solrConfig        the url under which the solrconfig.xml can be found
+     * @param solrSchema        the url under which the schema.xml can be found or null if the default schema should be used
+     */
+    public static void uploadConfiguration(int port, String configurationName, URL solrConfig, URL solrSchema) throws URISyntaxException, IOException {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("action", "UPLOAD");
+        parameters.put("name", configurationName);
+        HttpUrl url = generateSolrURL(port, Arrays.asList("admin", "configs"), parameters);
+
+        byte[] configurationZipFile = generateConfigZipFile(solrConfig, solrSchema);
+        executePost(url, configurationZipFile);
+
+    }
+
+    /**
+     * Creates a new collection
+     *
+     * @param port              The Port on which solr is running
+     * @param collectionName    the name of the collection which should be created
+     * @param configurationName the name of the configuration which should used to create the collection
+     *                          or null if the default configuration should be used
+     */
+    public static void createCollection(int port, String collectionName, String configurationName) throws URISyntaxException, IOException {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("action", "CREATE");
+        parameters.put("name", collectionName);
+        parameters.put("numShards", "1");
+        parameters.put("replicationFactor", "1");
+        parameters.put("wt", "json");
+        if (configurationName != null) {
+            parameters.put("collection.configName", configurationName);
+        }
+        HttpUrl url = generateSolrURL(port, Arrays.asList("admin", "collections"), parameters);
+        executePost(url, null);
+    }
+
+    private static void executePost(HttpUrl url, byte[] data) throws IOException {
+
+        RequestBody requestBody = data == null ?
+            RequestBody.create(MediaType.parse("text/plain"), "") :
+            RequestBody.create(MediaType.parse("application/octet-stream"), data);
+        ;
+
+        Request request = new Request.Builder()
+            .url(url)
+            .post(requestBody)
+            .build();
+        Response response = httpClient.newCall(request).execute();
```

Comment:
The `Response` will need to be closed to avoid a resource leak. Try-with-resources block, perhaps?

Reply:
As far as i read the code, you should only close the response if the body is not null.
Therefore i check if the body is not null and close it only if needed.